### ISSUE TITLE
Make an MDA be an object that's owned by pymmcore and users can register new ones

### ISSUE THREAD

**Issue by ianhi** [tlambert03/pymmcore-plus#101]
It's a bit awkward to have MMCore have methods like `cancel` and  properties like `pause` interspersed with other methods as it is not longer obvious that they are associated with MDAs. The same also goes for the signals, we've intermixed MDA signals with normal microscope control signals. 

What about having an `mda`  object on core that would handle all these things.

```python
# events
@core.mda.events.frameReady.connect
def newFrame(img, event):
    print(img.shape)

# cancelling/pausing
core.mda.cancel()
core.mda.toggle_pause()

# running mda - keep on core for reason below
core.run_mda(seq) # internally calls self.mda.run(seq)
```

This could also be a way towards having pluggable acquisition engines. By deafult we would load the current loop, but you could register your own engine with `core.register_mda_engine`

**Comment by tlambert03** [comment]
> What about having an mda object on core that would handle all these things.

love it

**Comment by ianhi** [comment]
fixed in #102
